docs/aws/audit/elbmonitoring/rules/classic_elb_listener_security_remediation.mdx

```diff
@@ -1,6 +1,239 @@
+
+4. **Enable Access Logs:**
+   - Go to the "Attributes" tab of your selected ELB.
+   - In the "Access logs" section, click on "Edit attributes".
+   - Enable access logging and specify an S3 bucket to store the logs.
+   - This will help you monitor and audit access to your ELB, ensuring that any insecure configurations can be quickly identified and addressed.
+
+By following these steps, you can ensure that your ELBs are configured securely, reducing the risk of misconfigurations that could lead to security vulnerabilities.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Elastic Load Balancers (ELBs) from having insecure configurations using AWS CLI, you can follow these steps:
+
+1. **Ensure HTTPS Listener:**
+   Make sure your ELB has an HTTPS listener to encrypt data in transit.
+
+   ```sh
+   aws elb create-load-balancer-listeners --load-balancer-name <your-load-balancer-name> --listeners "Protocol=HTTPS,LoadBalancerPort=443,InstanceProtocol=HTTP,InstancePort=80,SSLCertificateId=<your-certificate-arn>"
+   ```
+
+2. **Configure Security Groups:**
+   Attach a security group to your ELB that allows only necessary traffic (e.g., HTTPS on port 443).
+
+   ```sh
+   aws elb apply-security-groups-to-load-balancer --load-balancer-name <your-load-balancer-name> --security-groups <your-security-group-id>
+   ```
+
+3. **Enable SSL Policy:**
+   Apply a secure SSL policy to your ELB to ensure strong encryption.
+
+   ```sh
+   aws elb set-load-balancer-policies-of-listener --load-balancer-name <your-load-balancer-name> --load-balancer-port 443 --policy-names <your-ssl-policy-name>
+   ```
+
+4. **Enable Access Logs:**
+   Enable access logging for your ELB to monitor and audit traffic.
+
+   ```sh
+   aws elb modify-load-balancer-attributes --load-balancer-name <your-load-balancer-name> --load-balancer-attributes '{"AccessLog":{"Enabled":true,"S3BucketName":"<your-s3-bucket-name>","EmitInterval":5,"S3BucketPrefix":"<your-prefix>"}}'
+   ```
+
+Replace placeholders like `<your-load-balancer-name>`, `<your-certificate-arn>`, `<your-security-group-id>`, `<your-ssl-policy-name>`, and `<your-s3-bucket-name>` with your actual values. These steps will help ensure your ELB is configured securely.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elastic Load Balancers (ELBs) from having insecure configurations in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to ensure your ELBs are securely configured:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session to interact with AWS services.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>